src/Domain/Model/ValueObject/DateTimeRangeValeObject.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types=1);
+
+namespace PcComponentes\Ddd\Domain\Model\ValueObject;
+
+class DateTimeRangeValeObject implements ValueObject
+{
+    private DateTimeValueObject $startDate;
+    private DateTimeValueObject $endDate;
+
+    private function __construct(DateTimeValueObject $startDate, DateTimeValueObject $endDate)
+    {
+        $this->startDate = $startDate;
+        $this->endDate = $endDate;
+    }
+
+    public static function from(DateTimeValueObject $startDate, DateTimeValueObject $endDate)
+    {
+        if ($startDate >= $endDate) {
+            throw new \InvalidArgumentException('Start date can not be greater than the end date.');
+        }
+
+        return new self($startDate, $endDate);
+    }
+
+    public function startDate(): DateTimeValueObject
```

Comment:
Knowing that we are in the context of a date range.
I don't know if start**Date**() and end**Date**() is necessary or I would just leave it in start() and end().